kroxylicious-operator/pom.xml

```diff
@@ -148,11 +160,21 @@
             <artifactId>jackson-dataformat-yaml</artifactId>
             <scope>test</scope>
```

Comment:
does this need to be compile/runtime scoped now? kroxylicious-runtime is using `com.fasterxml.jackson.dataformat.yaml.YAMLFactory`

Reply:
That just means it's a transitive dependency of the operator. While we're not actually importing or referencing `YAMLWhatever` in the operator's `src/main/java ` then it needn't be a compile or runtime scope here.

Reply:
I guess it's not an issue until we assemble the dependencies, but currently if you assembled the `kroxylicious-operator` deps it would fail in `ProxyConfigSecret` when calling `ConfigParser.createObjectMapper()`. It works in tests because we've got the dep scoped to test.

Reply:
I'm not sure I understand the point you're making. The operator depends on the runtime at `compile` and the runtime depends on jackson yaml at `compile`. So the transitive dep is `compile`, and AFAICS the assembly plugin should grok that, no?

Reply:
the operator currently declares it at test scope which beats the transitive compile scope

```
kroxylicious-operator$ mvn org.apache.maven.plugins:maven-dependency-plugin:3.8.0:list | grep dataformat-yaml
[INFO]    com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.17.2:test -- module com.fasterxml.jackson.dataformat.yaml
```

Reply:
TIL.

The playing around I did earlier with the `maven-assembly-plugin` suggests it does actually do the "right thing" (i.e. it ended up in the dist), but then the fact that it's different from the dependency plugin is a sharp edge. 

Thanks!